src/main/java/kitchenpos/menus/tobe/application/MenuDisplaySyncService.kt

```diff
@@ -0,0 +1,29 @@
+package kitchenpos.menus.tobe.application
+
+import kitchenpos.menus.tobe.domain.Menu
+import kitchenpos.menus.tobe.domain.MenuRepository
+import kitchenpos.products.domain.ProductPriceChanged
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Propagation
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.transaction.event.TransactionPhase
+import org.springframework.transaction.event.TransactionalEventListener
+
+@Service
+class MenuDisplaySyncService(
+    private val menuRepository: MenuRepository,
+) {
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    @TransactionalEventListener(phase = TransactionPhase.AFTER_COMMIT)
+    fun syncMenusDisplayStatus(productPriceChanged: ProductPriceChanged) {
+        val menus = menuRepository.findAllByProductId(productPriceChanged.productId)
+
+        menus.forEach { syncMenuDisplayStatus(it) }
+    }
+
+    private fun syncMenuDisplayStatus(menu: Menu) {
+        if (!menu.isMenuPriceValid) {
+            menu.inActivateDisplayStatus()
+        }
+    }
```

Comment:
단순히 menu 객체에게 메시지를 보내도록 변경해보면 어떨까요? 🙂
```suggestion
    menu.syncMenuDisplayStatus();
```

Reply:
현재 변경된 productPrice로 동기화하는 부분이 누락된 것 같은데요. 
만약 이 기능이 추가된다면.  아래와 같이 menu 객체에게 'menuProductPrice` 같이 메시지를 보낼 수 있을 것 같은데요. 
이 메서드 안에서 가격을 변경한 뒤, 객체가 스스로 자신의 가격 유효성을 판별하여 메뉴의 전시 노출/ 미노출 상태를 적절히 변경할 수 있을 것 같아요. 🙂 

```kotlin
menu.changeMenuProductPrice(...)
```

Reply:
의견주신것처럼 메시지로 동기화하는게 내부구현도 숨기고 동기화하는 로직을 같이 넣어둘 수 있어서 좋을것같네요. 좋은 의견감사합니다

Reply:
`changeMenuProduct` 메서드에 대한 테스트도 추가해주시면 좋을 것 같아요. 🙂